src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -913,7 +913,17 @@ static void applyRepairCommandPoolSize(Config config)
     @VisibleForTesting
     static void applyTrackWarningsValidations(Config config)
     {
-        config.track_warnings.validate("track_warnings");
+        validateTrackWarnings("coordinator_read_size", config.coordinator_read_size_warn_threshold, config.coordinator_read_size_fail_threshold);
+        validateTrackWarnings("local_read_size", config.local_read_size_warn_threshold, config.local_read_size_fail_threshold);
+        validateTrackWarnings("row_index_size", config.row_index_size_warn_threshold, config.row_index_size_fail_threshold);
+    }
+
+    private static void validateTrackWarnings(String name, DataStorageSpec warn, DataStorageSpec fail)
```

Comment:
```suggestion
    private static void validateReadThresholds(String name, DataStorageSpec warn, DataStorageSpec fail)
```

Reply:
I guess this would also extend to things like `QueryOptions.TrackWarnings`? (i.e. We'd rename to `ReadThresholds` or something similar?)